flink-libraries/flink-cep/src/main/java/org/apache/flink/cep/operator/CepOperator.java

```diff
@@ -417,6 +405,9 @@ private void processEvent(NFAState nfaState, IN event, long timestamp) throws Ex
                             timestamp,
                             afterMatchSkipStrategy,
                             cepTimerService);
+            if (nfa.getWindowTime() > 0 && nfaState.isNewStartPartialMatch()) {
+                registerTimer(timestamp + nfa.getWindowTime());
```

Comment:
If the `windowTime` of the `NFA` equals to 0, should this registers a time for {@code current watermark + 1}?

Reply:
In my understanding, if there is a window in NFA then we need to register a timer to trigger this window. But if there is no window, what is the purpose of registering the timer of current watermark +1 here?